File: 3138/main.go
package main

import (
	"fmt"
)

func main() {
	fmt.Println(minAnagramLength("abba"))
}

func minAnagramLength(s string) int {
	n := len(s)
	check := func(m int) bool {
		var count0 [26]int
		for j := 0; j < n; j+= m {
			var count1 [26]int
			for k := j; k < j+m; k++ {
				count1[s[k] - 'a']++
			}
			if j > 0 && count0 != count1 {
				return false
			}
			count0 = count1
		}
		return true
	}
	for i := 1; i < n; i++ {
		if n % i != 0 {
			continue
		}
		if check(i) {
			return i
		}
	}
	return n
}

File: 3138/main.rs
struct Solution {}

impl Solution {
    pub fn min_anagram_length(s: String) -> i32 {
        let n = s.len();
        let check = |m: usize| -> bool {
            let mut count0 = vec![0; 26];
            for j in (0..n).step_by(m) {
                let mut c1 = vec![0; 26];
                for k in j..j+m {
                    c1[s.as_bytes()[k] as usize - b'a' as usize] += 1;
                }
                if j > 0 && count0 !=c1 {
                    return false;
                }
                count0 = c1;
            }
            true
        };
        for i in 1..n {
            if n % i != 0 {
                continue
            }
            if check(i) {
                return i as i32;
            }
        }
        n as i32
    }
}

fn main() {
    println!("{}", Solution::min_anagram_lenght("abba".to_string()));
}